# EPEK

## Contributors:


## Reference:
This repository provides the datasets and codes associated with the following research article:

Ananya Natarajan<sup>#</sup>, Nikhil Chivukula<sup>#</sup>, Gokul Balaji Dhanakoti, Ajaya Kumar Sahoo, Janani Ravichandran*, Areejit Samal*, [<i>EPEK: creation and analysis of an Ectopic Pregnancy Expression Knowledgebase</i>](https://www.sciencedirect.com/science/article/pii/S1476927123000579), Computational Biology and Chemistry, 104:107866, 2023.<br>
(<sup>#</sup> Joint First Authors, * Corresponding Authors)

## Repository Organization
This repository is organized into two folders:
- [EnrichmentAnalysis](https://github.com/asamallab/EPEK/tree/master/EnrichmentAnalysis)
- [Comparison](https://github.com/asamallab/EPEK/tree/master/Comparison)

### [EnrichmentAnalysis](https://github.com/asamallab/EPEK/tree/master/EnrichmentAnalysis):
This folder contains the code to plot for:
- [GO Term enrichment analysis](https://github.com/asamallab/EPEK/tree/master/EnrichmentAnalysis/GOTermEnrichment) (Figure 3)
- [Pathway enrichment analysis](https://github.com/asamallab/EPEK/tree/master/EnrichmentAnalysis/KEGGEnrichment) (Figure 4)
- [Disease enrichment analysis](https://github.com/asamallab/EPEK/tree/master/EnrichmentAnalysis/DiseaseEnrichment) (Figure 7)

### [Comparison](https://github.com/asamallab/EPEK/tree/master/Comparison):
This folder contains the code to generate UpSet plot (Figure 6).

Each folder has additional README files explaining the contents within them.

## Software requirements
Python package requirements:
pandas, seaborn, Matplotlib, NumPy

R package requirements:
UpSetR, SQLite, readxl, writexl

## References

[1] A. Natarajan, N. Chivukula, G.B. Dhanakoti, A.K. Sahoo, J. Ravichandran, A. Samal, EPEK: creation and analysis of an Ectopic Pregnancy Expression Knowledgebase, (2023). https://doi.org/10.1016/j.compbiolchem.2023.107866.

[2] S. Joseph, S.D. Mahale, Endometriosis Knowledgebase: a gene-based resource on endometriosis, Database. 2019 (2019) baz062. https://doi.org/10.1093/database/baz062.

[3] M. Sharma, I. Kundu, R.S. Barai, S. Bhaye, K. Desai, K. Pokar, S. Idicula-Thomas, Enrichment analyses of diseases and pathways associated with precocious puberty using PrecocityDB, Sci Rep. 11 (2021) 4203. https://doi.org/10.1038/s41598-021-83446-z.

[4] S.M. Agarwal, D. Raghav, H. Singh, G.P.S. Raghava, CCDB: a curated database of genes involved in cervix cancer, Nucleic Acids Research. 39 (2011) D975–D979. https://doi.org/10.1093/nar/gkq1024.

[5] M. Sharma, R.S. Barai, I. Kundu, S. Bhaye, K. Pokar, S. Idicula-Thomas, PCOSKBR2: a database of genes, diseases, pathways, and networks associated with polycystic ovary syndrome, Sci Rep. 10 (2020) 14738. https://doi.org/10.1038/s41598-020-71418-8.

[6] Y.A. Barbitoff, A.A. Tsarev, E.S. Vashukova, E.M. Maksiutenko, L.V. Kovalenko, L.D. Belotserkovtseva, A.S. Glotov, A Data-Driven Review of the Genetic Factors of Pregnancy Complications, IJMS. 21 (2020) 3384. https://doi.org/10.3390/ijms21093384.

[7] M. Kanehisa, Toward understanding the origin and evolution of cellular organisms, Protein Science. 28 (2019) 1947–1951. https://doi.org/10.1002/pro.3715.

[8] M. Kanehisa, M. Furumichi, Y. Sato, M. Kawashima, M. Ishiguro-Watanabe, KEGG for taxonomy-based analysis of pathways and genomes, Nucleic Acids Research. (2022) gkac963. https://doi.org/10.1093/nar/gkac963.

## Citation
In case you use the codes herein, please cite the following research article:

Ananya Natarajan<sup>#</sup>, Nikhil Chivukula<sup>#</sup>, Gokul Balaji Dhanakoti, Ajaya Kumar Sahoo, Janani Ravichandran*, Areejit Samal*, [<i>EPEK: creation and analysis of an Ectopic Pregnancy Expression Knowledgebase</i>](https://www.sciencedirect.com/science/article/pii/S1476927123000579), Computational Biology and Chemistry, 104:107866, 2023.
